File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        bool res = false;

        unordered_set<int> s;

        for(int i = 0; i < (int)nums.size(); i ++) {
            if(s.count(nums[i]) > 0) {
                // nums[i] is already in s
                res = true;
                break;
            }
            else {
                s.insert(nums[i]);
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = {} #count, num

        for num in nums:
            if num in d:
                d[num] += 1
            else:
                d[num] = 1
        
        hp = []

        for key in d:
            heapq.heappush(hp, (-d[key], key))
        
        res = []

        for i in range(k):
            res.append(heapq.heappop(hp)[1])
        
        return res

